src/tests/unit/platform/test_policy_ruleconfigs.py

```diff
@@ -0,0 +1,296 @@
+
+
+@pytest.mark.parametrize("new_data, get_id, handler, message", [
+    ({"id": "88b19232-7ebb-48ef-a198-2a75a282de5d", "name": "Privilege Escalation", "inherited_from": "",
+      "category": "core_prevention", "parameters": {"WindowsAssignmentMode": "BLOCK"}},
+     None, does_not_raise(), None),
+    ({"id": "88b19236-7ebb-48ef-a198-2a75a282de5d", "name": "Privilege Escalation", "inherited_from": "",
+      "category": "core_prevention", "parameters": {"WindowsAssignmentMode": "BLOCK"}},
+     "88b19232-7ebb-48ef-a198-2a75a282de5d", pytest.raises(InvalidObjectError),
+     "invalid rule config ID 88b19236-7ebb-48ef-a198-2a75a282de5d"),
+    ({"id": "88b19232-7ebb-48ef-a198-2a75a282de5d", "name": "Privilege Escalation", "inherited_from": "",
+      "category": "core_prevention", "parameters": {}},
+     None, does_not_raise(), None),
+    ({"id": "88b19232-7ebb-48ef-a198-2a75a282de5d", "name": "Privilege Escalation", "inherited_from": "",
+      "category": "core_prevention", "parameters": {"WindowsAssignmentMode": 666}},
+     None, pytest.raises(InvalidObjectError), "parameter error: 666 is not of type 'string'"),
+    ({"id": "88b19232-7ebb-48ef-a198-2a75a282de5d", "name": "Privilege Escalation", "inherited_from": "",
+      "category": "core_prevention", "parameters": {"WindowsAssignmentMode": "BOGUSVALUE"}},
+     None, pytest.raises(InvalidObjectError), "parameter error: 'BOGUSVALUE' is not one of ['REPORT', 'BLOCK']"),
+])
+def test_rule_config_validate_inside_policy(cbcsdk_mock, new_data, get_id, handler, message):
+    """Tests rule configuration validation when it's part of a policy."""
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65535/configs/presentation',
+                             POLICY_CONFIG_PRESENTATION)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65535, copy.deepcopy(FULL_POLICY_1), False, True)
+    rule_config_id = get_id if get_id is not None else new_data['id']
+    rule_config = policy.object_rule_configs[rule_config_id]
+    rule_config._info = copy.deepcopy(new_data)
+    with handler as h:
+        rule_config.validate()
+    if message is not None:
+        assert h.value.args[0] == message
+
+
+def test_rule_config_refresh(cbcsdk_mock):
+    """Tests the rule config refresh() operation."""
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536', FULL_POLICY_1)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, FULL_POLICY_1, False, True)
+    # Replace all rule configs with the base class for purposes of this test
+    cfgs = policy._info.get("rule_configs", [])
+    ruleconfigobjects = [PolicyRuleConfig(api, policy, cfg['id'], cfg, force_init=False, full_doc=True) for cfg in cfgs]
+    policy._object_rule_configs = dict([(rconf.id, rconf) for rconf in ruleconfigobjects])
+    policy._object_rule_configs_need_load = False
+    # proceed with test
+    rule_config = random.choice(list(policy.object_rule_configs.values()))
```

Comment:
I'm not a fan of the use of `random.choice` for testing as this produces potentially inconsistent testing results which may be hard to discover in the future

Reply:
This code should work on _any_ arbitrary rule config. I could set it up to always use the "first" one or whatever, but that runs the risk of special characteristics of that _particular_ rule config to cause the test to be passed while masking problems.

Reply:
However this could also produce random failures which may be hard to diagnose. Please find a more consistent and repeatable testing pattern

Reply:
How 'bout I just test _every_ rule configuration in those lists, just to make sure? It's actually almost as easy to write as just picking _one._

Reply:
That's better as the test would be consistent and repeatable